spynnaker/pyNN/utilities/bit_field_utilities.py

```diff
@@ -203,9 +222,28 @@ def write_bitfield_init_data(
         in_edge = proj._projection_edge
         if in_edge not in seen_app_edges:
             seen_app_edges.add(in_edge)
-            for machine_edge in in_edge.machine_edges:
-                if machine_edge.post_vertex.vertex_slice == vertex_slice:
-                    machine_edges.append(machine_edge)
+
+            # If neuromodulation is turned on on this edge then treat
+            # edges with reward or punishment receptors differently
+            post_vertex = in_edge.post_vertex
+            neuromodulation = in_edge.is_neuromodulated(post_vertex)
+
+            receptor_type = in_edge.synapse_information[0].receptor_type
+            if neuromodulation and (receptor_type == "reward" or
+                                    receptor_type == "punishment"):
```

Comment:
This should ask the synapse type whether the receptor type is neuromodulated

Reply:
This should probably also go through the synapse information rather than rely on the first as two projections with different parameters could mess this up!